nomenclature/testing.py

```diff
@@ -144,8 +145,12 @@ def assert_valid_structure(
             f"Definitions directory not found: {path / definitions}"
         )
 
-    if dimensions == ():  # if "dimensions" were not specified
-        dimensions = [x.stem for x in (path / definitions).iterdir() if x.is_dir()]
+    if not dimensions:  # if "dimensions" were not specified
+        dimensions = (
+            NomenclatureConfig.from_file(path / "nomenclature.yaml").dimensions
+            if (path / "nomenclature.yaml").is_file()
+            else [x.stem for x in (path / definitions).iterdir() if x.is_dir()]
+        )
```

Comment:
I'm wondering if could be replaced by 
```python
dsd = DataStructureDefiniton(definition, dimensions)
```
and then also pass `dsd` to the check-functions below instead of again parsing the *definitions* directory?